app/frontend/src/features/profile/hooks/useUpdateHostingPreferences.ts

```diff
@@ -33,7 +35,12 @@ export default function useUpdateHostingPreferences() {
       },
       onSuccess: () => {
         queryClient.invalidateQueries(["user", userId]);
-        history.push(userRoute);
+        if (user) {
+          history.push(routeToUser({
+            username: user.username,
+            tab: "home"
+          }));
+        }
```

Comment:
The user should exist here, even if the cache is stale. If for some reason it doesn't, it could make a really difficult to track bug ("Clicking save has no effect"). I think maybe better do `user!.username` - at least that way there'll be a console error (unfortunately it wouldn't trigger our error component since this is async).